### ISSUE TITLE
🐛 Bug Report: Nonce error 500 internal server error with social logins

### ISSUE THREAD

**Issue by dillfrescott** [stonith404/pingvin-share#584]
### 👟 Reproduction steps

Try to sign in with google or link microsoft account to account

### 👍 Expected behavior

It should work smoothly

### 👎 Actual Behavior

Error 500

### 📜 Logs

```
pingvin-share-1  |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
pingvin-share-1  |     at async OAuthController.callback (/opt/app/backend/dist/src/oauth/oauth.controller.js:50:22)
pingvin-share-1  | [Nest] 53  - 09/03/2024, 6:58:31 AM   ERROR [ExceptionsHandler] Cannot read properties of null (reading 'nonce')
pingvin-share-1  | TypeError: Cannot read properties of null (reading 'nonce')
pingvin-share-1  |     at MicrosoftProvider.getUserInfo (/opt/app/backend/dist/src/oauth/provider/genericOidc.provider.js:86:35)
pingvin-share-1  |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
pingvin-share-1  |     at async OAuthController.callback (/opt/app/backend/dist/src/oauth/oauth.controller.js:50:22)
pingvin-share-1  | [Nest] 53  - 09/03/2024, 6:59:09 AM     LOG [OAuthService] Successful login for user xdillfrescott@gmail.com from IP 172.20.0.1
pingvin-share-1  | [Nest] 53  - 09/03/2024, 6:59:09 AM   ERROR [ExceptionsHandler] Cannot read properties of null (reading 'totpVerified')
pingvin-share-1  | TypeError: Cannot read properties of null (reading 'totpVerified')
pingvin-share-1  |     at AuthService.generateToken (/opt/app/backend/dist/src/auth/auth.service.js:87:18)
pingvin-share-1  |     at OAuthService.signIn (/opt/app/backend/dist/src/oauth/oauth.service.js:67:30)
pingvin-share-1  |     at async OAuthController.callback (/opt/app/backend/dist/src/oauth/oauth.controller.js:57:27)
```

**Comment by stonith404** [comment]
I've improved the error logging for OIDC errors. Could you please try `v1.0.3` and share the logs again?

**Comment by Hennbaer** [comment]
Got a similar error with keycloak.

I am on 1.0.3 but it doesn't seem to log much more.

```
pingvin-share-1  | [Nest] 52  - 09/05/2024, 3:26:53 PM   ERROR [ExceptionsHandler] Cannot read properties of null (reading 'totpVerified')
pingvin-share-1  | TypeError: Cannot read properties of null (reading 'totpVerified')
pingvin-share-1  |     at AuthService.generateToken (/opt/app/backend/dist/src/auth/auth.service.js:87:18)
pingvin-share-1  |     at OAuthService.signIn (/opt/app/backend/dist/src/oauth/oauth.service.js:67:30)
pingvin-share-1  |     at async OAuthController.callback (/opt/app/backend/dist/src/oauth/oauth.controller.js:57:27)
```

The next lines show up in the log when refreshing the page after getting the 500 error (I've censored the request query):
```
pingvin-share-1  | [Nest] 52  - 09/05/2024, 3:26:53 PM   ERROR [OidcProvider] Can not get ID Token from response {
pingvin-share-1  |   "error": "invalid_grant",
pingvin-share-1  |   "error_description": "Code not valid"
pingvin-share-1  | }
pingvin-share-1  | [Nest] 52  - 09/05/2024, 3:26:53 PM   ERROR [OAuthExceptionFilter] Internal Server Error
pingvin-share-1  | [Nest] 52  - 09/05/2024, 3:26:53 PM   ERROR [OAuthExceptionFilter] Request query: {
pingvin-share-1  |   "state": "xxxxxxxxxxxxxxxx",
pingvin-share-1  |   "session_state": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
pingvin-share-1  |   "code": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
pingvin-share-1  | }
```
The pingvin "Oops something went wrong" page shows then.

Normal password login still works.

**Comment by Hennbaer** [comment]
Probably has something to do with 0.28.0. 
If I go back to 0.27.0 OIDC works again.

**Comment by stonith404** [comment]
@Hennbaer Thanks for your detailed description, I might have found a solution. Would you mind to test the `stonith404/pingvin-share:development` image and check if it works?

**Comment by Hennbaer** [comment]
Seems like it works.

I pulled the development docker image and I can log in via OpenID, no problem.

Good work!

**Comment by stonith404** [comment]
Okay great! This should be now fixed in `v1.0.4`.

**Comment by Hennbaer** [comment]
Thanks very much!
